src/util/math.h

```diff
@@ -38,40 +45,38 @@ inline bool even(T value) {
 #pragma intrinsic(fabs)
 #endif
 
-inline int roundUpToPowerOf2(int v) {
-    int power = 1;
-    while (power < v && power > 0) {
+// return value of 0 indicates failure (no greater power possible)
+constexpr unsigned int roundUpToPowerOf2(unsigned int v) {
+    const auto uv = static_cast<unsigned int>(v);
+#ifdef __cpp_lib_int_pow2
```

Comment:
```suggestion
#if (defined(__cpp_lib_int_pow2) && __cpp_lib_int_pow2 >= 202002L)
```

Reply:
Why are you checking for `__cpp_lib_int_pow2 >= 202002L`? I don't understand how this is necessary. Also, shouldn't be it be `__cpp_lib_int_pow2 == 202002L` instead?

Reply:
Because they may advance the version for future changes and than it will silently use our fallback path.

Reply:
Right, but they only advance it with significant changes. We could just add a static_assert that checks for `202002L`. I still don't understand why we check for the "version" at all.

Reply:
Because this define is already available in earlyer versions. From 202002 the function is available. It will unlikely disappear in future versions and when it does, build fails so we have the assertion anyway.

Reply:
> Because this define is already available in earlier versions


Mhmm. Thats confusing to me. I guess typical C++ :shrug: